### PR TITLE
BAI-1141 Experiment Ordering

### PR DESCRIPTION
**Pull request by bw27492**

### CODE REVIEW

File: lib/python/tests/test_model.py [hunk 1]
```diff
@@ -145,7 +145,21 @@ def test_publish_experiment_standard(standard_experiment):
...
+
+@pytest.mark.integration
+def test_publish_experiment_standard_ordered(standard_experiment):
+    standard_experiment.publish(mc_loc="performance.performanceMetrics", select_by="MAX:accuracy")
+
+    model_card = standard_experiment.model.model_card
+    model_card = NestedDict(model_card)
+    metrics_array = model_card[("performance", "performanceMetrics")][0]["datasetMetrics"]
+
+    expected_accuracy = 0.5
```
Comment (GB27247) [inline lib/python/tests/test_model.py:162]: Test for failure

File: lib/python/src/bailo/helper/model.py [hunk 2]
```diff
@@ -471,15 +471,24 @@ def from_mlflow(self, tracking_uri: str, experiment_id: str):
...
+        select_by: str | None = None,
+    ):
         """Publishes a given experiments results to the model card.
 
         :param mc_loc: Location of metrics in the model card (e.g. performance.performanceMetrics)
-        :param run_id: Local experiment run ID to be selected
         :param semver: Semantic version of release to create (if artifacts present), defaults to 0.1.0 or next
         :param notes: Notes for release, defaults to ""
+        :param run_id: Local experiment run ID to be selected, defaults to None
+        :param select_by: String describing experiment to be selected (e.g. "MIN|MAX:accuracy"), defaults to None
```
Comment (GB27247) [inline lib/python/src/bailo/helper/model.py:488]: Format with this in accordance with MLFlow

**Review by GB27247 — CHANGES_REQUESTED** [review]

File: lib/python/src/bailo/helper/model.py [hunk 3]
```diff
@@ -471,15 +471,24 @@ def from_mlflow(self, tracking_uri: str, experiment_id: str):
 
         logger.info(f"Successfully imported MLFlow experiment %s.", experiment_id)
 
-    def publish(self, mc_loc: str, run_id: str, semver: str = "0.1.0", notes: str = ""):
+    def publish(
+        self,
+        mc_loc: str,
+        semver: str = "0.1.0",
```
[the remark this answers is not in the document]
Comment (bw27492) [reply lib/python/src/bailo/helper/model.py:477]: This is needed

File: lib/python/src/bailo/helper/model.py (hunk 2 above)
Comment (bw27492) [reply lib/python/src/bailo/helper/model.py:488]: Done, now `accuracy MIN|MAX`

File: lib/python/tests/test_model.py (hunk 1 above)
Comment (bw27492) [reply lib/python/tests/test_model.py:162]: Done

**Review by GB27247 — APPROVED** [review]

**Pull request merged**